core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -161,42 +184,22 @@
       // invalid scheme. Just fall through and will try "dns:///foo.googleapis.com:8080"
       uriSyntaxErrors.append(e.getMessage());
     }
-    if (nameResolver == null) {
+    // It may be Opaque if the host was parsed as the scheme
+    if (targetUri == null || targetUri.isOpaque()) {
```

Comment:
An alternative could be the following conditional:

if (targetUri == null || targetUri.host() == null) {

If the host was accidentally parsed as a scheme, this would also check it.